src/System.Runtime.Extensions/src/System/Environment.Win32.cs

```diff
@@ -219,7 +219,7 @@ private static string GetKnownFolderPath(string folderGuid, SpecialFolderOption
             {
                 if (hr == Interop.Shell32.COR_E_PLATFORMNOTSUPPORTED)
                 {
-                    throw new PlatformNotSupportedException();
+                    throw new PlatformNotSupportedException(SR.NotSupported);
```

Comment:
This is likely a mistake made when following the original code. I assume this was related to the PAL not having a Unix implementation of this API- should just be removed. (Windows will never throw a COR_E hresult.)